flocker/node/agents/testtools/_blockdevice.py

```diff
@@ -891,117 +891,81 @@ class InvalidConfig(Exception):
+            'run storage backend functional tests.'
         )
 
     config_file_path = environ.get('FLOCKER_FUNCTIONAL_TEST_CLOUD_CONFIG_FILE')
     if config_file_path is None:
-        raise InvalidConfig(
-            'Supply the path to a cloud credentials file '
+        raise SkipTest(
+            'Supply the path to a backend configuration file '
             'using the FLOCKER_FUNCTIONAL_TEST_CLOUD_CONFIG_FILE environment '
-            'variable. See: '
-            'https://docs.clusterhq.com/en/latest/gettinginvolved/acceptance-testing.html '  # noqa
-            'for details of the expected format.'
+            'variable.'
+        )
+
+    # ie storage-drivers.rackspace
+    config_section = environ.get(
+        'FLOCKER_FUNCTIONAL_TEST_CLOUD_CONFIG_SECTION',
+    )
+    if config_section is None:
+        raise SkipTest(
+            'Supply the section of the config file '
+            'containing the configuration for the driver under test '
+            'with the FLOCKER_FUNCTIONAL_TEST_CLOUD_CONFIG_SECTION '
+            'environment variable.'
         )
 
     with open(config_file_path) as config_file:
         config = yaml.safe_load(config_file.read())
 
-    section = config.get(platform_name)
-    if section is None:
-        raise InvalidConfig(
-            "The requested cloud platform "
-            "was not found in the configuration file. "
-            "Platform: %s, "
-            "Configuration File: %s" % (platform_name, config_file_path)
-        )
-
-    provider_name = section.get('provider', platform_name)
-    try:
-        provider_environment = ProviderType.lookupByName(provider_name)
-    except ValueError:
-        raise InvalidConfig(
-            "Unsupported provider. "
-            "Supplied provider: %s, "
-            "Available providers: %s" % (
-                provider_name,
-                ', '.join(p.name for p in ProviderType.iterconstants())
-            )
-        )
+    section = None
+    for section in config_section.split('.'):
+        config = config.get(section)
 
-    if provider_environment != provider_type:
+    if config is None:
         raise InvalidConfig(
-            "The requested cloud provider (%s) is not the provider running "
-            "the tests (%s)." % (provider_type.name, provider_environment.name)
+            "The requested section "
+            "was not found in the configuration file. "
+            "Section: %s, "
+            "Configuration File: %s" % (config_section, config_file_path)
         )
 
-    # XXX - If provider type is Rackspace, we add the auth_plugin config
-    # here.  The CI system should be configured to provide this
-    # parameter as part of the config, to match the documented way of
-    # configuring Rackspace.
-    if provider_environment == ProviderType.rackspace:
-        section['auth_plugin'] = 'rackspace'
+    # XXX A hack to work around the fact that the sub-sections of
```

Comment:
@sarum90 I'll leave it in this branch and we can discuss removing it as soon as I land the segredos fix.